include/zephyr/drivers/timer/nrf_grtc_timer.h

```diff
@@ -0,0 +1,202 @@
+ *
+ * @param handler User function called in the context of the GRTC interrupt.
+ *
+ * @param user_data Data passed to the handler.
+ *
+ * @retval 0 if the compare channel was set successfully.
+ * @retval -EPERM if either channel is unavailable or SYSCOUNTER is not running.
+ */
+int z_nrf_grtc_timer_set(int32_t chan, uint64_t target_time,
+			 z_nrf_grtc_timer_compare_handler_t handler, void *user_data);
+
+/** @brief Abort a timer requested with z_nrf_grtc_timer_set().
+ *
+ * If an abort operation is performed too late it is still possible for an event
+ * to fire. The user can detect a spurious event by comparing absolute time
+ * provided in callback and a result of z_nrf_grtc_timer_read(). During this
+ * operation interrupt from that compare channel is disabled. Other interrupts
+ * are not locked during this operation.
+ *
+ * @param chan Channel ID between 1 and CONFIG_NRF_RTC_TIMER_USER_CHAN_COUNT.
+ */
+void z_nrf_grtc_timer_abort(int32_t chan);
+
+/** @brief Convert system clock time to GRTC ticks.
+ *
+ * @p t can be absolute or relative.
+ *
+ * @retval Positive value represents @p t in GRTC tick value.
+ * @retval -EINVAL if @p t is out of range.
+ */
+uint64_t z_nrf_grtc_timer_get_ticks(k_timeout_t t);
+
+/** @brief Prepare channel for timestamp capture.
+ *
+ * Use z_nrf_grtc_timer_capture_task_address_get() to determine the register
+ * address that is used to trigger capture.
+ *
+ * @note Capture and compare are mutually exclusive features - they cannot be
+ *       used simultaneously on the same GRTC channel.
+ *
+ * @param chan Channel ID.
+ *
+ * @retval 0 if the channel was successfully prepared.
+ * @retval -EPERM if either channel is unavailable or SYSCOUNTER is not running.
+ */
+int z_nrf_grtc_timer_capture_prepare(int32_t chan);
+
+/** @brief Read timestamp value captured on the channel.
+ *
+ * The @p chan must be prepared using z_nrf_grtc_timer_capture_prepare().
+ *
+ * @param chan Channel ID.
+ *
+ * @param captured_time Pointer to store the value.
+ *
+ * @retval 0 if the timestamp was successfully caught and read.
+ * @retval -EBUSY if capturing has not been triggered.
+ */
+int z_nrf_grtc_timer_capture_read(int32_t chan, uint64_t *captured_time);
+
+/** @brief Prepare GRTC as a source of wake up event and set the wake up time.
+ *
+ * @note Calling this function should be immediately followed by low-power mode enter
+ *		(if it executed successfully).
+ *
+ * @param wake_time_us Relative wake up time in microseconds.
+ *
+ * @retval 0 if wake up time was successfully set.
+ * @retval -EPERM if the SYSCOUNTER is not running.
+ * @retval -ENOMEM if no available GRTC channels were found.
+ * @retval -EINVAL if @p wake_time_us is too low.
+ */
+int z_nrf_grtc_wakeup_prepare(uint64_t wake_time_us);
+
+/**
+ * @brief       Initialize the GRTC clock timer driver from an application-
+ *              defined function.
+ *
+ * @return      0 on success.
+ *				Negative error code(s).
```

Comment:
The compliance check also fails on the last line here. But it looks like this part anyway requires reformatting.

Reply:
Should be fixed now

Reply:
Yes, regarding the compliance check, it is fixed. However, this section still could be formatted better, to be consistent with the rest of the file. But I'm not going to insist on this.